### PR TITLE
Apply dyn bgd bonus in computing guide_count

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

This is the partner to https://github.com/sot/proseco/pull/372, see that PR for more details.

Requires:
- https://github.com/sot/proseco/pull/372
- https://github.com/sot/chandra_aca/pull/124

## Interface impacts

None.

### Unit tests with branches mentioned above

- [x] Mac

Independent check of unit tests by Javier
- [x] Mac

### Functional tests

See this [demonstration and basic test](https://gist.github.com/taldcroft/e749159add9ca8f9a3b7733cc0374600) notebook.
See also the functional testing in https://github.com/sot/proseco/pull/372.

### CODE REVIEW

File: sparkles/core.py [hunk 1]
```diff
@@ -550,10 +550,19 @@ def __init__(self, *args, **kwargs):
...
         if self.guides is not None:
-            self.guide_count = guide_count(self.guides['mag'], self.guides.t_ccd)
+            mags = self.guides['mag']
+            t_ccds = np.full_like(mags, self.guides.t_ccd)
+            if self.dyn_bgd_n_faint > 0:
+                # Apply the dynamic background t_ccd bonus to the
+                # dyn_bgd_n_faint faintest stars. See:
+                # https://nbviewer.org/urls/cxc.harvard.edu/mta/ASPECT/ipynb/misc/guide-count-dyn-bgd.ipynb
+                mags = np.sort(mags)
+                n_faint = min(self.dyn_bgd_n_faint, len(t_ccds))
```
Comment (javierggt) [inline sparkles/core.py:560]: this is not needed. The following is perfectly fine:
```python
t_ccds[-self.dyn_bgd_n_faint:] += self.dyn_bgd_dt_ccd
```
you can try it yourself:
```python
In [1]: a = np.arange(6)

In [2]: a[-10:]
Out[2]: array([0, 1, 2, 3, 4, 5])
```

**Review by javierggt — COMMENTED** [review]
I don't see any test of the changes in this PR. Am I missing something?

The notebook linked in the description tests the underlying proseco changes, but I don't think it uses this PR. Am i wrong?

**Review by javierggt — APPROVED** [review]
I would still remove that spurious statement, but at least it doesn't hurt.

File: sparkles/core.py (hunk 1 above)
Comment (taldcroft) [reply sparkles/core.py:560]: I had some recollection of being bit by that, but either something changed (not likely) or I remembered wrong (likely). In any case I'll fix this.

**Comment by taldcroft** [comment]
@javierggt - Added test and improved the code per your suggestions. Thanks! I had been thinking the functional testing was enough but the specific unit test is indeed better.

**Pull request merged**